modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -1298,6 +1319,93 @@ private CompletableFuture<Boolean> onPrimaryExpired(PrimaryReplicaEventParameter
             return falseCompletedFuture();
         }
 
+        // TODO: move to Replica https://issues.apache.org/jira/browse/IGNITE-23750
+        private void registerFailoverCallback(TablePartitionId replicationGroupId) {
+            CompletableFuture<Replica> replicaFuture = replicaManager.replica(replicationGroupId);
+
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-23753
+            // For now we assume that it's a possible situation to get null future on node's stopp process, but it should be fixed with
+            // a correct and consistent replica's state.
+            if (replicaFuture == null) {
+                return;
+            }
+
+            replicaFuture.thenApply(electedPrimaryReplica -> {
+                onLeaderElectedFailoverCallback = (leaderNode, term) -> changePeersAndLearnersAsyncIfPendingExists(
+                        electedPrimaryReplica,
+                        replicationGroupId,
+                        term
+                );
+
+                return electedPrimaryReplica.raftClient().subscribeLeader(onLeaderElectedFailoverCallback);
```

Comment:
Am I right that it worst case we will send changePeersAndLearners twice: on changePeersAndLearnersAsyncIfPendingExists and onLeaderElectedFailoverCallback?

Reply:
I may be don't fully understend. In the code above `changePeersAndLearnersAsyncIfPendingExists` *is* the callback. I did it as the field because (yet) I need the lambda's object to unsubscribe. 

Also this callback doesn't send the new request, just straightforward call `changePeersAndLearners` inside.

But `TableManager` do it, and yes, in worse case `changePeersAndLearners` will be called twice: from the new request sent from `TableManager` and the call from callback on leader elected.